test/parallel/test-net-connect-buffer.js

```diff
@@ -87,6 +89,7 @@ tcp.listen(0, common.mustCall(function() {
     console.error('write cb');
     assert.ok(connected);
     assert.strictEqual(socket.bytesWritten, Buffer.from(a + b).length);
+    assert.strictEqual(socket.pending, false);
```

Comment:
Should we also add an assertion for `socket.pending === true` after the `'close'` event is emitted on the socket?